src/EditorFeatures/Core.Wpf/NavigateTo/NavigateToItemProvider.Searcher.cs

```diff
@@ -91,7 +91,18 @@ internal async Task SearchAsync()
                 }
                 finally
                 {
-                    _callback.Done();
+                    var service = _solution.Workspace.Services.GetService<IWorkspaceStatusService>();
+                    if (!await service.IsFullyLoadedAsync(_cancellationToken).ConfigureAwait(false) &&
+                        _callback is INavigateToCallback2 callback2)
```

Comment:
Should this do the IsNaviageToCallback2 check first since that's the cheaper one?

Reply:
sure.